components/hab/src/default_values.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This file contains the string defaults values as well as environment variable strings
+//! for use in the clap layer of the application. This is not the final form for defaults.
+//! Eventually this will likely reside in it's own crate and be composed of fully typed
+//! default values. But as a first step we need a spot to consolidate those values and help simplify
+//! some of the logic around them.
+
+// Supervisor
+pub const GOSSIP_DEFAULT_IP: &str = "0.0.0.0";
+pub const GOSSIP_DEFAULT_PORT: u16 = 9638;
+lazy_static! {
+    pub static ref GOSSIP_DEFAULT_ADDR: String =
+        { [GOSSIP_DEFAULT_IP, &GOSSIP_DEFAULT_PORT.to_string()].join(":") };
```

Comment:
I think this is a bit simpler/more idiomatic for rust:
```suggestion
        { format!("{}:{}", GOSSIP_DEFAULT_IP, GOSSIP_DEFAULT_PORT) };
```

Reply:
If you want to go even bigger, I think it would be a good idea to move the entire parsing here instead of doing this _ad hoc_:

```rust
    pub static ref GOSSIP_DEFAULT_ADDR: SocketAddrV4 = {
        SocketAddrV4::new(
            GOSSIP_DEFAULT_IP
                .parse()
                .expect("GOSSIP_DEFAULT_IP can not be parsed."),
            GOSSIP_DEFAULT_PORT,
        )
    };
    pub static ref GOSSIP_DEFAULT_ADDR_STRING: String = { GOSSIP_DEFAULT_ADDR.to_string() };
```